sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWriteRecordsInconsistent.java

```diff
@@ -0,0 +1,265 @@
+import org.joda.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A transform to write sharded records to BigQuery using the Storage API. This transform uses the
+ * default stream to write the records. Records written will show up in BigQuery immediately,
+ * however exactly once is not guaranteed - duplicates may appear in the output. For exactly-once
+ * writes, use {@link StorageApiWritesShardedRecords} or {@link StorageApiWriteUnshardedRecords}.
+ */
+@SuppressWarnings("FutureReturnValueIgnored")
+public class StorageApiWriteRecordsInconsistent<DestinationT, ElementT>
+    extends PTransform<PCollection<KV<DestinationT, byte[]>>, PCollection<Void>> {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(StorageApiWriteRecordsInconsistent.class);
+
+  private final StorageApiDynamicDestinations<ElementT, DestinationT> dynamicDestinations;
+  private final CreateDisposition createDisposition;
+  private final String kmsKey;
+  private final BigQueryServices bqServices;
+  private final Coder<DestinationT> destinationCoder;
+  private static final ExecutorService closeWriterExecutor = Executors.newCachedThreadPool();
+
+  private static final Cache<String, StreamAppendClient> APPEND_CLIENTS =
+      CacheBuilder.newBuilder()
+          .expireAfterAccess(5, TimeUnit.MINUTES)
+          .removalListener(
+              (RemovalNotification<String, StreamAppendClient> removal) -> {
+                @Nullable final StreamAppendClient streamAppendClient = removal.getValue();
+                // Close the writer in a different thread so as not to block the main one.
+                runAsyncIgnoreFailure(closeWriterExecutor, streamAppendClient::close);
+              })
+          .build();
+
+  // Run a closure asynchronously, ignoring failures.
+  private interface ThrowingRunnable {
+    void run() throws Exception;
+  }
+
+  private static void runAsyncIgnoreFailure(ExecutorService executor, ThrowingRunnable task) {
+    executor.submit(
+        () -> {
+          try {
+            task.run();
+          } catch (Exception e) {
+            //
+          }
+        });
+  }
+
+  public StorageApiWriteRecordsInconsistent(
+      StorageApiDynamicDestinations<ElementT, DestinationT> dynamicDestinations,
+      CreateDisposition createDisposition,
+      String kmsKey,
+      BigQueryServices bqServices,
+      Coder<DestinationT> destinationCoder) {
+    this.dynamicDestinations = dynamicDestinations;
+    this.createDisposition = createDisposition;
+    this.kmsKey = kmsKey;
+    this.bqServices = bqServices;
+    this.destinationCoder = destinationCoder;
+  }
+
+  @Override
+  public PCollection<Void> expand(PCollection<KV<DestinationT, byte[]>> input) {
+    String operationName = input.getName() + "/" + getName();
+    // Append records to the Storage API streams.
+    return input.apply(
+        "Write Records",
+        ParDo.of(new WriteRecordsDoFn(operationName))
```

Comment:
is the difference between the inconsistent and consistent writes the finalization of streams? (e.g. like here: https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWriteUnshardedRecords.java#L103)

Doesn't it make sense to reuse the logic from that file? I see this file has metrics, so that's nice... just wondering. WDYT?

Reply:
StorageApiWriteUnshardedRecords implements a fairly intricate protocol to ensure exactly-once writes to BQ. It involves shuffles to ensure consistent mapping of records to streams, coordination of stream creation, appends, and flushes, and a protocol to throw away and recreate streams to prevent duplicates on certain errors.

This PR adds a much simpler transform that does not guarantee exactly once. No shuffle is needed, and the default Storage API stream is used. None of the flushing or stream creation/recreation is needed either. I don't think it would be easy to use that other file, though there might be some code snippets in both files that could be refactored into a helper class.

Reply:
I think this is a good idea actually. Trying it out now.

Reply:
Ok - this is now simply delegating to the class you suggested (which has been modified to support the default stream). Thanks for the suggestion!